### PR TITLE
Add org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

### PR DESCRIPTION
**Pull request by Bleuzen**
### Please confirm your submission meets all the criteria

- [ ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ ] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ ] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

### CODE REVIEW

**Comment by Bleuzen** [comment]
DO NOT MERGE!

This is very wip, still in testing stage.
The plan is to package all plugins from Chowdhury-DSP in one flatpak.
This would supersede the current https://github.com/flathub/org.freedesktop.LinuxAudio.Plugins.ChowTapeModel flatpak.

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by hfiguiere** [comment]
you should use the same approach as for the WhiteElephantAudio plugins, ie not a third party git with submodules.

**Comment by Bleuzen** [comment]
> you should use the same approach as for the WhiteElephantAudio plugins

While possible I try to make something easy to package, so that also other packagers besides on flathub have an easier time. This is simplified by having only one repo to build.

> ie not a third party git with submodules.

If this works well this could in some form land in the official ChowDSP releases repo. Third party is only for testing now, will be replaced later.

**Comment by hfiguiere** [comment]
the thing is that your way of doing it limit who can update the versions of the plugins to you. And require extra steps.

**Comment by Bleuzen** [comment]
> the thing is that your way of doing it limit who can update the versions of the plugins to you. And require extra steps.

This is only temporary for testing. Plan is to merge it upstream and the official plugin developer will make new tags and keep it updated.

**Comment by hfiguiere** [comment]
but to be fair, "all in one" packages like that might be the wrong approach.

**Comment by hfiguiere** [comment]
(unless this is upstream choice)

**Comment by Bleuzen** [comment]
> but to be fair, "all in one" packages like that might be the wrong approach.

I would like to see those plugins packaged in more repos. Lowering the bar and having something simple seems a good start so more packagers can pick it up without too much of an effort. 
Of course this can be discussed if you still think there is something wrong with it.



> (unless this is upstream choice)

Will see, we are still testing how this could work:

https://github.com/Chowdhury-DSP/releases/issues/1

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
build should be fixed now, however the first one is still running for some reason despite that it ran into an error. Not sure why it doesn't end. Is there a way to stop frozen builds? @hfiguiere

**Comment by Bleuzen** [comment]
@hfiguiere  any idea why the cloning fails now? A `git clone --recursive https://github.com/Chowdhury-DSP/releases.git` works here, but not in flatpak.

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
bot, stop

**Comment by Bleuzen** [comment]
ehhw :/

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by Bleuzen** [comment]
@hfiguiere @barthalion hm it fails at cloning the RTNeural submodule, but only on flathub. Cloning and build works just fine here locally. Any idea?

**Comment by hfiguiere** [comment]
no idea why this is failing. It looks like the git clone is too shallow.

**Comment by hfiguiere** [comment]
I think I changed my mind on all-in-one packages. I submitted GuitarML that way. This will limit the number of plugin packages.

So anything I can help with to get this in?

I'll submit BYOD (not in here) separately though because I have a version with the standalone version. It make sense to have it.

**Comment by hfiguiere** [comment]
Albeit it might make more sense to build the plugins one by one instead of using upstream meta project.

**Comment by hfiguiere** [comment]
See https://github.com/Bleuzen/flathub/pull/1

**Comment by hfiguiere** [comment]
Any chance you can merge the PR so we can move forward with this submission?

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by hfiguiere** [comment]
uh, it did build locally :-/ Need figure that part out.

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by hfiguiere** [comment]
this repo that fail is also the same that's used for https://github.com/flathub/org.freedesktop.LinuxAudio.Plugins.ChowTapeModel/

**Comment by hfiguiere** [comment]
@barthalion any idea of why it fails to fetch a submodule of a submodule. Build works locally and the same repo (and commit) works in a different package.

(also I'm not the OP so I can't really try things like disable-shallow)

**Comment by hfiguiere** [comment]
BYOD is submitted separately for the standalone app https://github.com/flathub/flathub/pull/3918

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

**Comment by hfiguiere** [comment]
looks like this is working better.

**Comment by hfiguiere** [comment]
BYOD is available on flathub as both standalone and plugin built from https://github.com/flathub/com.chowdsp.BYOD

**Comment by hfiguiere** [comment]
shall we move this out of draft?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins

File: org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins.json [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                "ChowKick_LV2",
+                "ChowKick_VST3"
+            ],
+            "post-install": [
+                "install -d ${FLATPAK_DEST}/lv2",
+                "cp -R ./ChowKick_artefacts/Release/LV2/*.lv2 ${FLATPAK_DEST}/lv2/",
+                "strip ${FLATPAK_DEST}/lv2/ChowKick.lv2/*.so",
+                "install -d ${FLATPAK_DEST}/vst3",
+                "cp -R ./ChowKick_artefacts/Release/VST3/*.vst3 ${FLATPAK_DEST}/vst3/",
+                "strip ${FLATPAK_DEST}/vst3/ChowKick.vst3/Contents/*-linux/*.so",
```
[the remark this answers is not in the document]
Comment (Bleuzen) [reply org.freedesktop.LinuxAudio.Plugins.ChowDSP-Plugins.json:172]: thanks :)

**Comment by hfiguiere** [comment]
@barthalion  it's far in the backlog, but this should be ready now,

**Pull request closed without merging**